### PR TITLE
ci: release artifacts

### LINKED ISSUE #1665: Upload war and jar artifacts to the releases

### LINKED ISSUE #1667: Adjust the test workflows for the jar deployment

### PR DESCRIPTION
**Pull request by takb**
### Information about the changes
- default build option is JAR
- added workflow triggered on release to build and attach JAR and WAR artifacts

Fixes #1665 .
Fixes #1667 .

### CODE REVIEW

File: .github/workflows/vulnerability-scanning.yml [hunk 1]
```diff
@@ -54,8 +54,10 @@ jobs:
       - name: Build war file
```
Comment (koebi) [inline .github/workflows/vulnerability-scanning.yml]: This is building the .jar and .war-file, so its name should reflect that.

**Review by koebi — CHANGES_REQUESTED** [review]
LGTM, one minor nitpick.

File: .github/workflows/vulnerability-scanning.yml (hunk 1 above)
Comment (TheGreatRefrigerator) [reply .github/workflows/vulnerability-scanning.yml]: Done

**Comment by TheGreatRefrigerator** [comment]
Vulnerabilitiy and docker issues will be handled separately.

**Review by koebi — APPROVED** [review]

**Pull request merged**